### ISSUE TITLE
"TypeError: Cannot read property 'displayName' of undefined"

### ISSUE THREAD

**Issue by catherineomega** [kitajchuk/node-squarespace-server#84]
When I run the server, it lets me log in, then fails here:
```
> sqs-server: Running site on localhost:5050
> sqs-server: AUTH - Login to Squarespace!
> sqs-server: GET - /
> sqs-server: TEMPLATE - front.region
> sqs-server: QUERY - insights
```

("front.region" doesn't exist in my template;  'insights' is the name of the blog feature that DOES exist.)

The error returned:
```
   TypeError: Cannot read property 'displayName' of undefined
    at sqsFormatters.item-classes (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/formatters.js:62:53)
    at /usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/formatters.js:172:23
    at _DoSubstitute (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/jsontemplate.js:420:13)
    at _Execute (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/jsontemplate.js:402:7)
    at Template.render (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/jsontemplate.js:798:3)
    at Template.expand (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/jsontemplate.js:803:8)
    at Object.render (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/squarespace-jsont.js:29:25)
    at handleDone (/usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:388:36)
    at handleQueried (/usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:401:13)
    at /usr/local/lib/node_modules/node-squarespace-server/squarespace-template.js:471:21
```
Any ideas? sqs is at 0.1.28 and Node is at v0.10.26.

**Comment by kitajchuk** [comment]
Oh yes, the template key is maybe confusing. The template it declares is not an actual template file in your theme. The server parses your local copy and creates a reference object for what template(s) to use and/or put together and use for rendering. My guess is you have a layout in your template.conf file called "front" which is why you would get that template key logging out.

Ok, aside from that, my first suggestion would be to ensure none of the issues here are yours:
https://github.com/kitajchuk/node-squarespace-server#troubleshooting

If you rule those out then let me know. Looking at your error, it has to do with the "item-classes" formatter. It appears that the author is undefined on the json context for that formatter when it gets called in your template. I'm not sure how that can happen. It may be an edge case, or it may be due to one of the above troubleshooting issues not being resolved.

Let me know what you find.

**Comment by catherineomega** [comment]
Aha, okay! That makes a lot more sense. Yes, you're correct, I have a "front"  in "layouts", and if I remove or rename the collection "insights" in the template, it will query the correspondingly collection--or not, if it's removed. (I've checked through the "insights" the client has posted, and they all appear to actually have authors as per the Squarespace UI, at least. And it happens on every page I've checked, regardless of template or regions it's using.)

I've also confirmed that I'm using https:// in siteurl, and that "sandbox": true--though that evidently makes no difference. 

But that said, I notice that the line that's mentioned first in formatters.js in node-squarespace-jsont (line 62) doesn't end in a semicolon: https://github.com/kitajchuk/node-squarespace-jsont/blob/master/lib/formatters.js. Adding one didn't do anything. Removing the entire line resulted in line 63 being singled out:
```
TypeError: Cannot call method 'toLowerCase' of undefined
    at sqsFormatters.item-classes (/usr/local/lib/node_modules/node-squarespace-server/node_modules/node-squarespace-jsont/lib/formatters.js:63:66)
```

But then when I remove line 63 as well... my page appears. So at least there's that? I don't really understand why those parts of val. aren't working, but at least I can get the mobile version of this site mocked up ASAP.
